indra/llmessage/llpacketring.cpp

```diff
@@ -43,329 +43,301 @@
 #include "message.h"
 #include "u64.h"
 
-///////////////////////////////////////////////////////////
-LLPacketRing::LLPacketRing () :
-    mUseInThrottle(false),
-    mUseOutThrottle(false),
-    mInThrottle(256000.f),
-    mOutThrottle(64000.f),
-    mActualBitsIn(0),
-    mActualBitsOut(0),
-    mMaxBufferLength(64000),
-    mInBufferLength(0),
-    mOutBufferLength(0),
-    mDropPercentage(0.0f),
-    mPacketsToDrop(0x0)
+constexpr S16 MAX_BUFFER_RING_SIZE = 1024;
+constexpr S16 DEFAULT_BUFFER_RING_SIZE = 256;
+
+LLPacketRing::LLPacketRing ()
+    : mPacketRing(DEFAULT_BUFFER_RING_SIZE, nullptr)
 {
+    LLHost invalid_host;
+    for (size_t i = 0; i < mPacketRing.size(); ++i)
+    {
+        mPacketRing[i] = new LLPacketBuffer(invalid_host, nullptr, 0);
+    }
 }
```

Comment:
This is almost a complete rewrite of LLPacketRing.

Previously it wasn't actually a ring although it had a seldom-used API that made it kinda look like a ring.  I'm purging that seldom-used stuff since it was only ever used for simulating slow network connections and these days there are better (external) tools available to force the viewer into low-bandwidth environments.

Going forward `LLPacketRing` has a real ring of pre-allocated `LLPacketBuffers` that can be used to buffer inbound UDP packets that we don't yet have the per-frame time to process.